components/Tag/Tag.tsx

```diff
@@ -0,0 +1,31 @@
+import { FC } from "react";
+import styles from "./Tag.module.css";
+
+interface TagTypeProp {
+  typeName: string;
+}
+
+const Tag: FC<TagTypeProp> = ({ typeName }) => {
+  let tagBgColor = "";
+
+  switch (typeName) {
```

Comment:
todo: Instead of a switch-case, you can use an object to map the background. It's easier to write and faster. Let me show you an example where we map sizes to colors:

```js
const sizeColors = {
  big: "red",
  medium: "yellow",
  small: "green",
};

const tagColor = sizeColors["big"];

```

Now let's see it applied to this project with TS:


<details>



```ts

// Type all possible Holidays
type Holidays =
  | "National holiday"
  | "Observance"
  | "Common local holiday"
  | "Season";

// Type Props
interface TagProp {
  typeName: Holidays;
}

// Do the mapper
const holidayBgMapper: {
  [key in Holidays]: string;
} = {
  "National holiday": styles.tag_national_holiday,
  Observance: styles.tag_observance,
  "Common local holiday": styles.tag_common_holiday,
  Season: styles.tag_season,
};

const Tag: FC<TagProp> = ({ typeName }) => {
  // Access directly the bg by object key:value
  const tagBgColor = holidayBgMapper[typeName];

  return (
    <div className={`${styles.tag_container} ${tagBgColor}`}>{typeName}</div>
  );
};



```
</details>

Reply:
Good ideas

Reply:
this one I am learning now